### PR TITLE
Adds the Prometheus <-> Maas functionality

### PR DESCRIPTION
**Pull request by xgerman**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,8 +1,13 @@
-FROM stackbrew/ubuntu:trusty
+FROM stackbrew/ubuntu:xenial
 
 
 RUN apt-get update && apt-get -y upgrade && apt-get -y install curl
 RUN curl https://monitoring.api.rackspacecloud.com/pki/agent/linux.asc | apt-key add - 
 RUN echo "deb http://stable.packages.cloudmonitoring.rackspace.com/ubuntu-14.04-x86_64 cloudmonitoring main" > /etc/apt/sources.list.d/rackspace-monitoring-agent.list
```
Comment (johnsom) [inline Dockerfile]: Should this change to 16.04 since you moved to xenial?

File: check_prometheus.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+    status_ok(m_name=check_name)
+
+def main(args):
+    check(args)
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description='Retrieve values for a check from the Prometheus API')
+    parser.add_argument('prometheus_endpoint', nargs='?',
```
Comment (johnsom) [inline check_prometheus.py]: Doesn't '?' require a default? I can't remember.

File: check_prometheus.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        status_err(traceback.format_exc(), force_print=True, exception=e,
+                   m_name='maas')
+    else:
+        if STATUS:
+            print(STATUS)
+        for metric in METRICS:
+            print(metric)
+
+def check(args):
+    check_name = 'maas_k8s_prometheus_%s' % check
```
Comment (johnsom) [inline check_prometheus.py:76]: Seems odd to hard code the "maas_k8s_" here when the script name is more generic. Shouldn't the full string come in from the args?

File: check_prometheus.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    try:
+        r = requests.get('%s//api/v1/query' % args.prometheus_endpoint,
```
Comment (johnsom) [inline check_prometheus.py]: Why '//' here?

File: check_prometheus.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+                         params={'query': args.query},
+                         timeout=5)
```
Comment (johnsom) [inline check_prometheus.py]: You might want to add this as an optional argument.

File: check_prometheus.py (hunk 3 above)
Comment (xgerman) [reply check_prometheus.py:76]: this is meant for multiple checks so only checks vary but that will always be maaas_prometheus

**Pull request merged**